pkgs/applications/misc/blender/default.nix

```diff
@@ -101,10 +102,30 @@ stdenv.mkDerivation (finalAttrs: {
   pname = "blender";
   version = "4.1.0";
 
-  src = fetchurl {
-    url = "https://download.blender.org/source/${finalAttrs.pname}-${finalAttrs.version}.tar.xz";
-    hash = "sha256-3AAtguPDQMk4VcZoRzDQGAG2aaKbHMa3XuuZC6aecj8=";
-  };
+  srcs = [
+    (fetchgit {
+      name = "source";
+      url = "https://projects.blender.org/blender/blender.git";
+      rev = "v${finalAttrs.version}";
+      hash = "sha256-4LtBOufcaUJP49QxT9/EcFVjNbyvGf/gYFoVtljod1Y=";
+      fetchLFS = true;
+    })
+    (fetchFromGitea {
+      name = "addons";
+      domain = "projects.blender.org";
+      owner = "blender";
+      repo = "blender-addons";
+      rev = "v${finalAttrs.version}";
+      hash = "sha256-OHEUtGiubZNf3nsT4LfHuGyOhfVPk2H3osZ1lWJL4jI=";
+    })
+  ];
+
+  sourceRoot = "source";
+
+  prePatch = ''
+    mkdir scripts/addons
+    cp -Rv --no-preserve=all ../addons/* scripts/addons
+  '';
```

Comment:
```suggestion
  srcs = [
    (fetchFromGitea {
      inherit domain owner;

      repo = "blender";
      rev = "v${finalAttrs.version}";
      hash = "sha256-Adpn9ueRWHoH+PZLtH2m/HwulyGTkxVXF6jnEncO+Bk=";
    })
    (fetchFromGitea {
      name = "addons";
      inherit domain owner;

      repo = "blender-addons";
      rev = "v${finalAttrs.version}";
      hash = "sha256-OHEUtGiubZNf3nsT4LfHuGyOhfVPk2H3osZ1lWJL4jI=";
    })
    (fetchFromGitea {
      name = "addons-contrib";
      inherit domain owner;

      repo = "blender-addons-contrib";
      rev = "v${finalAttrs.version}";
      hash = "sha256-XbwtpMoD1GSbogQ7uIxtnoi1vRjOi6DvTTI7GqhlkSw=";
    })
    (fetchFromGitea {
      name = "assets";
      inherit domain owner;

      repo = "blender-assets";
      rev = "v${finalAttrs.version}";
      hash = "sha256-AkN+VpwPlWCXYtsfjzC99gqupzh96tT7VuTRQjkPBDg=";
    })
  ];

  sourceRoot = "source";

  prePatch = ''
    mkdir scripts/addons scripts/addons_contrib
    cp -rv --no-preserve=all ../addons/* scripts/addons
    cp -rv --no-preserve=all ../addons-contrib/* scripts/addons_contrib
    cp -rv --no-preserve=all ../assets/working/* release/datafiles/assets
  '';
```

This is to grab all sources without the use of LFS

Reply:
Yeah, I am. I didn't really understand it before, but Blender now uses both. 

as per https://projects.blender.org/blender/blender/commit/3dc832a90415635c8387ba641d7722a9fd1e236c, they switched from SVN to git submodules with LFS.

Reply:
Again, this is just an alternative, if `fetchgit` isn't really desired.

Reply:
Another advantage to this approach is that we can control which version of the addons, test etc. are included.